### ISSUE TITLE
Add Query Pattern Clustering Analysis

### ISSUE THREAD

**Issue by evanvolgas** [ashita-ai/conduit#45]
## Overview
Implement clustering analysis on query embeddings to identify routing patterns and optimization opportunities.

## Background
With embeddings database (Issue #43), we can cluster similar queries and analyze routing performance per cluster.

### 1. Pattern-Based Routing Rules
- Cluster: "Math/calculation queries" → Route to claude-3-5-sonnet (high accuracy)
- Cluster: "Simple definitions" → Route to gpt-4o-mini (cost efficiency)
- Cluster: "Creative writing" → Route to gpt-4o (quality)

### 2. Cluster Performance Analysis
```python
# Identify underperforming clusters
clusters = await analyzer.cluster_queries(min_cluster_size=50)
for cluster in clusters:
    regret = cluster.avg_regret_vs_oracle
    if regret > 0.15:  # 15% worse than Oracle
        print(f"Cluster {cluster.id} needs attention")
        print(f"Sample queries: {cluster.samples[:5]}")
        print(f"Current routing: {cluster.model_distribution}")
```

### 3. Novel Query Detection
- Alert when queries don't fit existing clusters
- Trigger exploration strategy for novel patterns
- Track cluster drift over time

### 1. Clustering Service
```python
# conduit/analysis/clustering.py
from sklearn.cluster import DBSCAN
import numpy as np

class QueryClusterer:
    async def cluster_queries(
        self,
        time_range: str = "last_week",
        min_samples: int = 10,
        eps: float = 0.15  # embedding distance threshold
    ) -> list[QueryCluster]:
        """Cluster queries using DBSCAN on embeddings."""
        # Fetch embeddings from pgvector
        embeddings = await db.fetch_embeddings(time_range)
        
        # DBSCAN clustering
        clustering = DBSCAN(eps=eps, min_samples=min_samples, metric='cosine')
        labels = clustering.fit_predict(embeddings)
        
        # Analyze each cluster
        clusters = []
        for label in set(labels):
            cluster = await self._analyze_cluster(label, embeddings, labels)
            clusters.append(cluster)
        
        return clusters
```

### 2. Cluster Performance Metrics
```python
@dataclass
class QueryCluster:
    id: int
    size: int
    sample_queries: list[str]
    centroid: np.ndarray
    
    # Routing performance
    model_distribution: dict[str, float]
    avg_quality: float
    avg_cost: float
    avg_latency: float
    regret_vs_oracle: float
    
    # Trend analysis
    growth_rate: float  # cluster size change over time
    drift_score: float  # how much centroid moves
```

### 3. Cluster-Based Routing Hints
```python
# conduit/engines/router.py
async def route_with_cluster_hints(self, query: Query) -> Decision:
    """Route using cluster patterns as prior knowledge."""
    
    # Find nearest cluster
    cluster = await clusterer.find_nearest_cluster(query.embedding)
    
    if cluster and cluster.confidence > 0.8:
        # Use cluster's best model as hint
        best_model = cluster.best_performing_model
        # Bias bandit toward this model (Bayesian prior)
        decision = await bandit.select_arm(
            features=query.features,
            prior_model=best_model
        )
    else:
        # Standard routing
        decision = await bandit.select_arm(query.features)
    
    return decision
```

### 4. Cluster Dashboard Endpoint
```python
# conduit/api/routes.py
@app.get("/api/clusters")
async def get_clusters(time_range: str = "last_week"):
    """Get query cluster analysis."""
    clusters = await clusterer.cluster_queries(time_range)
    return {
        "total_clusters": len(clusters),
        "clusters": [c.to_dict() for c in clusters],
        "outliers": sum(1 for c in clusters if c.id == -1)
    }
```

## Acceptance Criteria
- [ ] DBSCAN clustering implementation
- [ ] Cluster performance metrics
- [ ] Cluster-based routing hints (optional Bayesian prior)
- [ ] Novel query detection
- [ ] Cluster drift tracking
- [ ] API endpoint for cluster analysis
- [ ] >90% test coverage
- [ ] Documentation in README.md

## Effort Estimate
**Medium** (10-14 hours)
- Clustering implementation: 3-4 hours
- Performance metrics: 2-3 hours
- Routing hints integration: 2-3 hours
- API endpoints: 2-3 hours
- Testing and docs: 1-2 hours

## Dependencies
- Embeddings database (Issue #43)
- scikit-learn (already installed)
